spec.md

```diff
@@ -1081,6 +1081,88 @@ $instance = new class extends \Foo implements
 };
 ```
 
+## 9. Attributes
+
+### 9.1 Basics
+
+Attribute names must immediately follow the opening attribute block indicator `#[` with no space.
+
+If an attribute has no arguments, the `()` MUST be omitted.
+
+The closing attribute block indicator `]` MUST follow the last character of the attribute name or the closing `)` of
+its argument list, with no preceding space.
+
+### 9.2 Placement
+
+Attributes on classes, methods, functions, constants and properties MUST
+be placed on their own line, immediately prior to the structure being described.
+
+For attributes on parameters, if the parameter list is presented on a single line,
+the attribute MUST be placed inline with the parameter it describes, separated by a single space.
+If the parameter list is split into multiple lines for any reason, the attribute MUST be placed on
+its own line prior to the parameter, indented the same as the parameter.
+
+If a comment docblock is present on a structure that also includes an attribute, the comment block MUST
+come first, followed by any attributes, followed by the structure itself.  There MUST NOT be any blank lines
+between the docblock and attributes, or the attributes and the structure.
+
+If two separate attribute blocks (denoted by separate `#[]` markers) are used in a multi-line context,
+they MUST be on separate lines with no blank lines between them.
+
+### 9.3 Compound attributes
+
+Multiple attributes MAY be placed in the same attribute block (`#[]`) if and only if the entire block is listed on a
+single line.  They must be separated by a comma with a space following but no space preceding.  If the attribute list
```

Comment:
```suggestion
come first, followed by any attributes, followed by the structure itself. There MUST NOT be any blank lines
between the docblock and attributes, or the attributes and the structure.

If two separate attribute blocks (denoted by separate `#[]` markers) are used in a multi-line context,
they MUST be on separate lines with no blank lines between them.

### 9.3 Compound attributes

Multiple attributes MAY be placed in the same attribute block (`#[]`) if and only if the entire block is listed on a
single line. They must be separated by a comma with a space following but no space preceding. If the attribute list
```

Reply:
2 spaces after a period is fine.

Reply:
I agree with @mathroc, we should prefer consistency in the document wherever possible.